### ISSUE TITLE
Supporting Yggdrasil Network

### ISSUE THREAD

**Issue by immartian** [isontheline/pro.webssh.net#1073]
# Feature description

Yggdrasil is a unique networking layer that creates a decentralized, encrypted network. By using WebSSH in conjunction with Yggdrasil, users can establish secure, direct connections to their home computers or other devices on the Yggdrasil network. This setup is particularly beneficial because it eliminates the need for dynamic DNS (Domain Name System) services or complex network configurations.

Key Benefits:

	1.	Ease of Access: Users can easily access their remote systems from anywhere, as long as they have a web browser and internet connectivity. This simplifies the process of managing and accessing remote devices, especially when users are on the go.
	2.	Enhanced Security: By leveraging Yggdrasil’s encrypted network, WebSSH ensures that the remote login process is secure and private. The encryption provided by Yggdrasil adds an extra layer of security, protecting the data transmitted during the SSH session.
	3.	No Need for Dynamic DNS: Typically, accessing a home computer remotely requires setting up dynamic DNS to handle changing IP addresses. However, with Yggdrasil, users can directly connect to their devices without this setup, as Yggdrasil handles the addressing and routing within its network.
	4.	Decentralization: Yggdrasil’s decentralized nature means that users don’t rely on central servers or services to connect to their devices. This increases reliability and resilience, as the network does not have single points of failure.

fyi, more on https://yggdrasil-network.github.io/2018/07/15/remote-access.html

# Screenshots / Draws
just put a checkbox 
- [x] Turn On Yggdrasil Network 
would be enough on UI

**Comment by isontheline** [comment]
Hello @immartian 👋

Thank you so much for reaching me out 🙏

I don't think it's just a "toggle to turn on" in this case as yggdrasil network will require a Network Extension inside WebSSH.

WebSSH is not a VPN application, so it could be rejected by Apple when reviewing.

**Comment by immartian** [comment]
Thanks, @isontheline , for your prompt feedback and great program. I've been a loyal user of WebSSH since its early days, now a pro user. 

I've been using both Yggdrasil across my projects and devices as well. I understand @neilalexander made it as an experiment, but IMO, it's a great idea and should be put into protocol level. 

I agreed that it may need Apple's approval to get this layer integrated into WebSSH although I'm not sure it's the reason the test IOS app of Ygg didn't continue. But I think for the use case of WebSSH, Ygg has more reason to be a reason to add on. Maybe it's even possible to create a new narrative, "Connect back to your home server from anywhere on your phone". 

Anyway, just think about it and maybe we can have a chat sometime.

**Comment by isontheline** [comment]
I also wonder why @neilalexander project is down.

Last week I’ve been required to remove VPN-Over-SSH from the macOS release.

I’m now reading other Internet to find documentation about how I will transform VPN-Over-SSH in a near future.

**Comment by immartian** [comment]
right. although it's pretty promising idea, that project didn't take off but went wild.
